src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -98,7 +107,33 @@ public override bool Execute()
             }
 
             ConfigFileContents = builder.ToString();
-            return true;
+            return WriteMSBuildEditorConfig();
+        }
+
+        public bool WriteMSBuildEditorConfig()
```

Comment:
Consider making this logic conditional on actually having a `FileName` value, so that in tests we can just look at the `ConfigFileContents`